core/src/test/scala/org/ensime/indexer/FileWatcherSpec.scala

```diff
@@ -18,11 +19,11 @@ import org.scalatest.tagobjects.Retryable
 import org.scalatest.time._
 
 sealed trait FileWatcherMessage
-final case class Added(f: FileObject) extends FileWatcherMessage
-final case class Removed(f: FileObject) extends FileWatcherMessage
-final case class Changed(f: FileObject) extends FileWatcherMessage
-final case class BaseAdded(f: FileObject) extends FileWatcherMessage
-final case class BaseRemoved(f: FileObject) extends FileWatcherMessage
+final case class Added(f: EnsimeFile) extends FileWatcherMessage
```

Comment:
these look like they should be `RawFile`.

Remember, `EnsimeFile` indicates that it is a file or a path within an archive. So we want to be as specific as possible where we can both in terms of what the inputs for a method are but also for the outputs. This helps us know what we actually need to implement.